src/NexusMods.App.UI/Pages/LoadoutPage/LoadoutItemModel.cs

```diff
@@ -47,8 +55,23 @@ public LoadoutItemModel(LoadoutItemId loadoutItemId)
             return ids.Select(id => (Id: id, ShouldEnable: shouldEnable)).ToArray();
         });
 
-        _modelActivationDisposable = WhenModelActivated(this, static (model, disposables) =>
+        _modelActivationDisposable = WhenModelActivated(this, (model, disposables) =>
         {
```

Comment:
Keep the `static` you have access to the class instance and can pass anything else as state into the lambda.

Reply:
The `state` parameter is constrained by `where T : IReactiveR3Object`.

In order to pass the variables via state, I'd need to persist the remaining constructor fields to the class; however those fields are only used during initialisation. 

It can be solved by wrapping `this` in a struct, and delegating the whole interface implementation to what's under the hood, but then I'd be getting complaints about unnecessary boilerplate in the code. Or do a temporary field but that would be a waste of memory. Can't really win here.

Reply:
You can just use `WhenActivated` instead of `WhenModelActivated`.

Reply:
Alright, done, I actually didn't spot the other overload earlier, despite it being right below the other method.
Works for me, and is preferable.